src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -220,6 +223,21 @@ protected void AddMapInternal(string collection, string subCollecction, Indexing
             funcs.Add(map);
         }
 
+        internal void CheckDepthOfStackInOutputMap(IndexDefinition indexMetadata, DocumentDatabase documentDatabase)
+        {
+            var performanceHintConfig = documentDatabase.Configuration.PerformanceHints;
+            if (StackSizeInSelectClause > performanceHintConfig.MaxDepthOfRecursionInLinqSelect)
+            {
+                documentDatabase.NotificationCenter.Add(PerformanceHint.Create(
+                    documentDatabase.Name,
+                    $"Index '{indexMetadata.Name}' contains a lot of `let` clause.",
+                    $"Each of let clause is nesting projections and could potentially leads to StackoverflowException.",
+                    PerformanceHintType.Indexing,
+                    NotificationSeverity.Info,
+                    nameof(IndexCompiler)));
+            }
```

Comment:
Can we also put that warning into the log?

Reply:
done

Reply:
Add the number here.
And the amount you exceeded it by.

Do we have a way to add a link to docs?

Reply:
Included. 
About the docs: maybe in `msg` part?

Reply:
First we need to write about it in the docs. Then we can use the docs link generator and add it to the message.